pkg/controllers/node/termination/controller.go

```diff
@@ -77,13 +78,24 @@ func (c *Controller) Finalize(ctx context.Context, node *v1.Node) (reconcile.Res
 	if !controllerutil.ContainsFinalizer(node, v1beta1.TerminationFinalizer) {
 		return reconcile.Result{}, nil
 	}
+
+	nodeGracePeriodExpirationTime, err := c.terminationGracePeriodExpirationTime(ctx, node)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("couldn't evaulate node's terminationGracePeriod, %w", err)
+	} else if nodeGracePeriodExpirationTime != nil && time.Now().After(*nodeGracePeriodExpirationTime) {
+		logging.FromContext(ctx).Infof("node's terminationGracePeriod has expired, adding non-graceful shutdown taint: %v", node.Name)
+		if err := c.terminator.Taint(ctx, node, v1beta1.DisruptionNonGracefulShutdown); err != nil {
+			return reconcile.Result{}, fmt.Errorf("error while tainting node with node.kubernetes.io/out-of-service=nodeshutdown:NoExecute, %w", err)
```

Comment:
```suggestion
			return reconcile.Result{}, fmt.Errorf("error while tainting node with v1beta1.DisruptionNonGracefulShutdown.String(), %w", err)
```

I wonder if we can create a helper or stringify the value to make it easy to print the taint in a readable format. I think there may be another place in the Karpenter scheduler code that enables something like this. We should consider using it if we can.